### PR TITLE
pref: lift up coverage rate

### PR DESCRIPTION
**Pull request by orzyyyy**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [x] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

Related to #14775

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/upload/UploadList.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ const previewFile = (file: File | Blob, callback: Function) => {
   reader.readAsDataURL(file);
 };
 
-const extname = (url: string) => {
+export const extname = (url: string) => {
```
Comment (afc163) [inline components/upload/UploadList.tsx]: 最好不要为了测试暴露这些方法，去 https://jestjs.io/ 找找办法。

File: components/upload/UploadList.tsx [hunk 2]
```diff
@@ -72,6 +62,16 @@ export default class UploadList extends React.Component<UploadListProps, any> {
...
+  // https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL
+  previewFile = (file: File | Blob, callback: Function) => {
+    if (file.type && !imageTypes.includes(file.type)) {
+      callback('');
+    }
+    const reader = new FileReader();
+    reader.onloadend = () => callback(reader.result);
+    reader.readAsDataURL(file);
+  };
+
```
Comment (afc163) [inline components/upload/UploadList.tsx]: 放进来的意义是啥，`previewFile` 里没用到 this。

Comment (zy410419243) [reply components/upload/UploadList.tsx]: 为了方便测试跟不暴露方法...

Comment (zy410419243) [reply components/upload/UploadList.tsx]: 这写法有啥问题吗，有的话我看看怎么改...

Comment (afc163) [reply components/upload/UploadList.tsx]: 没啥大问题，如果是 eslint 会报错，目前 ts 没这个规则。另外本质上也是通过类方法对外暴露了。

**Review by ztplz — APPROVED** [review]

**Pull request merged**